classes/CartRule.php

```diff
@@ -857,21 +897,16 @@ protected function checkProductRestrictions(Context $context, $return_products =
         return (!$display_error) ? true : false;
     }
 
-    protected static function array_uintersect($array1, $array2)
-    {
-        $intersection = array();
-        foreach ($array1 as $value1) {
-            foreach ($array2 as $value2) {
-                if (CartRule::array_uintersect_compare($value1, $value2) == 0) {
-                    $intersection[] = $value1;
-                    break 1;
-                }
-            }
-        }
-        return $intersection;
-    }
-
-    protected static function array_uintersect_compare($a, $b)
```

Comment:
I assumed by the name of the function that it is compatible with the native `array_uintersect`, so it has been replaced with the native version.